improver_tests/wxcode/wxcode/test_WeatherSymbols.py

```diff
@@ -630,56 +663,328 @@ def test_old_naming_convention(self):
+                            atol=0,
+                        ),
+                    ),
+                    ">=",
+                    0.5,
+                ],
+            ],
+            "OR",
+        ]
         plugin = WeatherSymbols()
-        diagnostics = [
-            "probability_of_rainfall_rate_above_threshold",
-            "probability_of_lwe_snowfall_rate_above_threshold",
+        result = plugin.evaluate_condition_chain(self.cubes, chain)
+        c1 = chain[0][0][0]
+        c2 = chain[0][1][0]
+        expected = (self.cubes.extract(c1)[0].data >= 0.5) | (
+            self.cubes.extract(c2)[0].data >= 0.5
+        )
+        self.assertArrayEqual(result, expected)
+
+    def test_with_operators(self):
+        """Test a condition chain where the expressions contain operators."""
+        t = AuxCoord(0.1, units="mm hr-1")
+        t.convert_units("m s-1")
+        chain = [
+            [
+                [
+                    [
+                        iris.Constraint(
+                            name="probability_of_rainfall_rate_above_threshold",
+                            rainfall_rate=lambda cell: np.isclose(
+                                cell.point,
+                                t.points[0],
+                                rtol=WeatherSymbols().float_tolerance,
+                                atol=0,
+                            ),
+                        ),
+                        "-",
+                        iris.Constraint(
+                            name="probability_of_lwe_snowfall_rate_above_threshold",
+                            lwe_snowfall_rate=lambda cell: np.isclose(
+                                cell.point,
+                                t.points[0],
+                                rtol=WeatherSymbols().float_tolerance,
+                                atol=0,
+                            ),
+                        ),
+                    ],
+                    ">=",
+                    0.5,
+                ],
+                [
+                    iris.Constraint(
+                        name="probability_of_lwe_sleetfall_rate_above_threshold",
+                        lwe_sleetfall_rate=lambda cell: np.isclose(
+                            cell.point,
+                            t.points[0],
+                            rtol=WeatherSymbols().float_tolerance,
+                            atol=0,
+                        ),
+                    ),
+                    ">=",
+                    0.5,
+                ],
+            ],
+            "OR",
         ]
-        thresholds = [AuxCoord(0.03, units="mm hr-1"), AuxCoord(0.03, units="mm hr-1")]
-        result = plugin.construct_extract_constraint(diagnostics, thresholds, False)
+        plugin = WeatherSymbols()
+        result = plugin.evaluate_condition_chain(self.cubes, chain)
+        c1 = chain[0][0][0]
+        c2 = chain[0][1][0]
+        expected = (self.cubes.extract(c1)[0].data >= 0.5) | (
```

Comment:
I'm wondering if this test is correct. It seems the same as the one on line 819 in `test_basic`. I think it's missing the subtraction operator somehow.

Reply:
Yep, this was a copy-paste error!